### PR TITLE
Run only one instance each for send/recv loop

### PR DESCRIPTION
**Pull request by losfair**
TODO:

1. Messages might get randomly dropped now if sent quickly.

### CODE REVIEW

File: network/builder_test.go [hunk 1]
```diff
@@ -161,6 +161,34 @@ func TestWindowSize(t *testing.T) {
...
+	t.Parallel()
+
+	writeBufferSize := 2048
+	builder := NewBuilderWithOptions(
+		WriteBufferSize(writeBufferSize),
+	)
+	net, err := builder.Build()
+	if err != nil {
+		t.Errorf("Build() = %+v, expected <nil>", err)
+	}
```
Comment (doug-perlin) [inline network/builder_test.go]: can simply these nil checks in one line with
```
assert.Equal(t, nil, err)
```

File: network/network.go [hunk 2]
```diff
@@ -1,51 +1,39 @@
...
-	"github.com/perlin-network/noise/protobuf"
 
+	"bufio"
 	"github.com/gogo/protobuf/proto"
 	"github.com/gogo/protobuf/types"
 	"github.com/golang/glog"
 	"github.com/pkg/errors"
 	"github.com/xtaci/kcp-go"
-	"github.com/xtaci/smux"
+	"time"
```
Comment (doug-perlin) [inline network/network.go]: the order of the imports look strange

File: network/stream.go [hunk 3]
```diff
@@ -5,82 +5,89 @@ import (
...
 	}
 
-	// Flush writer.
-	err = writer.Flush()
-	if err != nil {
-		return errors.Wrap(err, "failed to flush writer")
+	for totalBytesWritten < len(buffer) && err == nil {
+		bytesWritten, err = w.Write(buffer[totalBytesWritten:])
+		if err != nil {
+			glog.Errorf("stream: failed to write entire buffer, err: %+v\n", err)
```
Comment (doug-perlin) [inline network/stream.go]: Is it okay to keep writing or should we break out of the loop on error and retry from the start

Comment (jack0) [reply network/stream.go]: well it will break out due to the err == nil in the if check

**Pull request merged**